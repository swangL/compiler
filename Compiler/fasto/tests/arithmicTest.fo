fun int main () =
  // negate
  let w = write(~10) in
  let w = write(~~10) in
  let w = write(~10 + 20) in

  // multiply
  let w = write(5 * 5) in 
  let w = write(0 * 5) in
  let w = write(5 * ~5) in
  let w = write(~5 * ~5) in
  let w = write(~5 * 0) in
  // multiply precedence
  let w = write(5 + 5 * 5 + 10 * 10) in

  // divide
  let w = write(10 / 2) in
  let w = write(~10 / 2) in
  let w = write(~10 / ~10) in
  let w = write(0 / 10) in
  // divide precedence
  let w = write(5 + 5 / 5 + 10 / 10) in
  let w = write(10 / 0) in // this should be last test because it throws error
                           // if the line number changes, pls update error msg in output file
  0