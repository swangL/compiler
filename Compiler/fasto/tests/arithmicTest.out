-10
10
10
25
0
-25
25
0
130
5
-5
1
0
7
Error: Division by zero at line 23